src/Workspaces/MSBuild/Test/NewlyCreatedProjectsFromDotNetNew.cs

```diff
@@ -220,13 +222,29 @@ static void TryCopyGlobalJson(string outputDirectory)
         async Task AssertProjectLoadsCleanlyAsync(string projectFilePath, string[] ignoredDiagnostics)
         {
             using var workspace = CreateMSBuildWorkspace();
-            var project = await workspace.OpenProjectAsync(projectFilePath, cancellationToken: CancellationToken.None);
+
+            string binlogPath;
+            if (Environment.GetEnvironmentVariable("HELIX_DUMP_FOLDER") is string helixDumpFolder && !string.IsNullOrEmpty(helixDumpFolder))
+                binlogPath = Path.Combine(helixDumpFolder, Path.ChangeExtension(Path.GetFileName(projectFilePath), ".binlog"));
+            else
+                binlogPath = Path.ChangeExtension(projectFilePath, ".binlog");
+
+            LoggerFactory.CreateLogger(nameof(CreateNewProject)).LogInformation($"Creating binlog at {binlogPath}");
+
+            var logger = new Microsoft.Build.Logging.BinaryLogger { Parameters = binlogPath };
+            var project = await workspace.OpenProjectAsync(projectFilePath, msbuildLogger: logger, cancellationToken: CancellationToken.None);
+            var analyzerLoadDiagnostics = new List<AnalyzerLoadFailureEventArgs>();
+
+            foreach (var analyzerReference in project.AnalyzerReferences.OfType<AnalyzerFileReference>())
+                analyzerReference.AnalyzerLoadFailed += (sender, e) => analyzerLoadDiagnostics.Add(e);
```

Comment:
We should delete this when we merge https://github.com/dotnet/roslyn/pull/80109.